pkg/cmd/stack/deploy.go

```diff
@@ -441,7 +441,11 @@ func getFailedStatusError(ctx context.Context, stack *model.Stack, svcName strin
 		for _, cStatus := range p.Status.ContainerStatuses {
 			totalRestarts += int(cStatus.RestartCount)
 		}
-		if totalRestarts >= maxRestartsToConsiderFailed {
+		maxSvcRestarts := stack.Services[dependingSvc].BackOffLimit
+		if maxSvcRestarts == 0 {
+			maxSvcRestarts = maxRestartsToConsiderFailed
+		}
+		if int32(totalRestarts) >= maxSvcRestarts {
```

Comment:
if maxSvcRestarts is not int32 ? will this be compared?

Reply:
We should either cast `totalRestarts` to `int32` or `maxSvcRestarts` to `int`. I casted `totalRestarts` because it's more explicit, but since both of them are integers they will get compared